Refactor toolresult

asr2003 opened the pull request.

asr2003 committed:
Update Cargo.toml

asr2003 committed:
Merge branch 'antinomyhq:main' into refactor-toolresult

asr2003 committed:
refactor: standardize tool responses to front matter format

autofix-troubleshooter commented:
Hi! I'm the <a href="https://autofix.ci"><img alt="autofix logo" src="https://autofix.ci/logo/logo.svg" width="16"><strong>autofix.ci</strong></a> troubleshooter bot.

It looks like you correctly set up a CI job that uses the autofix.ci GitHub Action, but the [**autofix.ci GitHub App**](https://github.com/apps/autofix-ci/) has not been installed for this repository. This means that autofix.ci unfortunately does not have the permissions to fix this pull request. If you are the repository owner, please install the app and then restart the CI workflow! 😃

asr2003 committed:
refactor: standardize tool responses to front matter format

asr2003 committed:
refactor: standardize tool responses to front matter format

asr2003 commented:
Details
Introduced a new rule for detecting when a Redpanda node becomes isolated (heartbeats fail) and triggers a Raft re-election, indicating quorum loss and leader election.

Added relevant log patterns that capture:

RPC timeouts (rpc::errc::exponential_backoff)

Disconnected endpoints (rpc::errc::disconnected_endpoint(node down))

Raft vote replies and “becoming the leader” messages

Updated category and tags to include high-availability and Raft/leader-election context.

Test Environment
Reproducible test setup (Maintainers invited):
[Redpanda 3-node Docker Compose](https://github.com/vectorizedio/redpanda/tree/dev/docker)
(Adapted from the official Vectorized examples to bring up 3 brokers on a private bridge network.)

Live CRE link:
CRE Playground Link

Check here for Sample Logs

Sample Detected Patterns
log
Copy
Edit
[2025-06-06T05:50:30] cluster - Error occurred while sending node status request: rpc::errc::exponential_backoff
[2025-06-06T06:27:17] r/heartbeat - Received error when sending heartbeats to node 1 - rpc::errc::exponential_backoff
[2025-06-06T06:27:31] cluster - unable to get node health report from 1 - rpc::errc::disconnected_endpoint(node down), marking node as down
[2025-06-06T06:28:43] raft - vote_stm.cc:77 - vote reply from {…} - vote_granted: 1, log_ok:1
[2025-06-06T06:28:43] raft - vote_stm.cc:264 - becoming the leader term:2
Reproduction Steps
bash
Copy
Edit
# 1) Bring up a 3-node Redpanda cluster
git clone https://github.com/vectorizedio/redpanda.git
cd redpanda/docker
docker-compose -f docker-compose-three-nodes.yml up -d

# 2) Let brokers converge and elect a controller
sleep 20

# 3) Simulate node isolation (stop broker on node 2)
docker stop redpanda2

# 4) Watch for detected patterns in broker-0’s logs
docker logs redpanda1 --tail 50 | grep -E "rpc::errc::exponential_backoff|disconnected_endpoint|vote_stm"

# 5) Observe Raft re-election:
#    - You should see “vote reply … vote_granted” followed by “becoming the leader term…” lines
Closes #59
/claim #59

asr2003 commented:
does it expands striclty iwth macro usingg 1;1 if not fic chec if any exosotng macros can be used or wirte one tp match 1;1
